x/auth/client/rest/sign.go

```diff
@@ -30,21 +26,27 @@ func SignTxRequestHandlerFn(cdc *codec.Codec, cliCtx context.CLIContext) http.Ha
 	return func(w http.ResponseWriter, r *http.Request) {
 		var m SignBody
 
-		body, err := ioutil.ReadAll(r.Body)
-		if err != nil {
+		if err := utils.ReadRESTReq(w, r, cdc, &m); err != nil {
 			utils.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
 			return
 		}
-		err = cdc.UnmarshalJSON(body, &m)
-		if err != nil {
+
+		// validate request
+		if !m.ValidateBasic(w) {
+			return
+		}
+
+		// validate tx
+		// discard error if it's CodeUnauthorized as the tx comes with no signatures
+		if err := m.Tx.ValidateBasic(); err != nil && err.Code() != sdk.CodeUnauthorized {
```

Comment:
Why do we have a special check for sdk.CodeUnauthorized?

Reply:
We're calling `ValidateBasic()` on a _unsigned_ Tx. It fails because of lack of sigs